DatePicker_NE_120224

nihalerdal opened the pull request:
Implemented a CustomDatePicker component using MUI. 
Updated the code replacing deprecated props.

rebs1203 commented on src/DatePicker.jsx:
```diff
@@ -0,0 +1,26 @@
+import { useState } from "react";
+
+const CustomDatePicker = ({ selectedDateRange, onRangeChange }) => {
+  const [dateRange, setDateRange] = useState(selectedDateRange || [null, null]);
```
Here instead of `[null, null]` you can just have it as `null`

jgoldenberg29 replied:
we typically want our state variables to maintain the same variable type so maybe we should go with an empty array? I think empty arrays are now registered as a falsey value in javascript (recent update). So it should work like null  as Rebecca suggested. [null, null] however, because it is an array that technically has a non-zero length would be a truthy value and might cause some issues.

rebs1203 commented on src/DatePicker.jsx:
```diff
@@ -0,0 +1,26 @@
+import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
+import { useState } from "react";
+
+const CustomDatePicker = ({ selectedDateRange, onRangeChange }) => {
```
Where are the params being passed?

nihalerdal replied:
I've also updated App.jsx file. Could you please try again?

rebs1203 commented on src/DatePicker.jsx:
```diff
@@ -0,0 +1,26 @@
+
+  const handleChange = (newDateRange) => {
+    setDateRange(newDateRange);
+    onRangeChange(newDateRange); //callback func to pass the date to the parent comp
```
Where can I see this function?

nihalerdal replied:
I've also updated App.jsx file. Could you please try again?

rebs1203 reviewed (commented):
Overall good job! Love seeing the MUI usage!

jgoldenberg29 reviewed (changes requested):
Let's try and keep our logic and state variables out of our app component. The app component is going to be almost exclusively for routing. Since our application is getting more complex, we are going to have many files that will look similar to our app from our todo list where they initialize state variables and define or import functions for sending requests and handling other complex logic.

nihalerdal commented:
The changes have been made. Please feel free to review it again

nihalerdal commented:
The date picker integration has been completed and tested! Thank you so much for sharing your feedback!

- Integrated the date picker component.
- Refactored the date picker to format dates in YYYY-MM-DD, aligning with the backend's expected format.
- Added logic to automatically fill the missing date if only one date is provided.
- Updated package-lock.json.
- Prevented to select dates earlier than today

StuvaScript reviewed (approved):
Looks great! I like the placement and how you can't pick past dates.

jgoldenberg29 reviewed (approved):
Great work Nihal! looks great and functions great!
